tests/io/v0_0_0/test_caching_backend.py

```diff
@@ -0,0 +1,153 @@
+import contextlib
+import json
+import os
+import random
+import socket
+import subprocess
+import sys
+import time
+import unittest
+
+import numpy as np
+import requests
+import skimage.io
+
+import slicedimage
+
+pkg_root = os.path.abspath(os.path.join(os.path.dirname(__file__), "..", "..", ".."))  # noqa
+sys.path.insert(0, pkg_root)  # noqa
+
+from tests.utils import TemporaryDirectory
+
+
+class TestCachingBackend(unittest.TestCase):
+    def setUp(self, timeout_seconds=5):
+        self.contexts = []
+        self.tempdir = TemporaryDirectory()
+        self.contexts.append(self.tempdir)
+        self.port = _unused_tcp_port()
+
+        if sys.version_info[0] == 2:
+            module = "SimpleHTTPServer"
+        elif sys.version_info[0] == 3:
+            module = "http.server"
+        else:
+            raise Exception("unknown python version")
+
+        self.contexts.append(ContextualChildProcess(
+            [
+                "python",
+                "-m",
+                module,
+                str(self.port),
+            ],
+            cwd=self.tempdir.name,
+        ).__enter__())
+
+        end = time.time() + timeout_seconds
+        while True:
+            try:
+                requests.get("http://0.0.0.0:{port}".format(port=self.port))
+                break
+            except requests.ConnectionError:
+                if time.time() > end:
+                    raise
+
+    def tearDown(self):
+        for context in self.contexts:
+            context.__exit__(*sys.exc_info())
+
+    def test_cached_backend(self):
+        """
+        Generate a tileset consisting of a single TIFF tile.  Deposit it where the HTTP server can
+        find the tileset, and fetch it.
+        """
+        # write the tiff file
+        data = np.random.randint(0, 65535, size=(100, 100), dtype=np.uint16)
```

Comment:
This looks good; consider also testing a larger file since we know that it behaves slightly differently.